tests/test_parser/test_properties/test_init.py

```diff
@@ -404,14 +404,23 @@ def test_property_from_data_str_enum(self, enum_property_factory, config):
             "ParentAnEnum": prop,
         }
 
+    @pytest.mark.parametrize(
+        "desc,extra_props",
+        [
+            ("3_1_implicit_type", {}),
+            ("3_1_explicit_type", {"type": ["string", "null"]}),
+            ("3_0_implicit_type", {"nullable": True}),
+            ("3_0_explicit_type", {"type": "string", "nullable": True}),
+        ],
+    )
     def test_property_from_data_str_enum_with_null(
```

Comment:
The reason that this unit test wasn't catching the problem was that there several ways the same nullable enum could be declared, and it was only testing one of them.